test/integration/httpApi.test.js

```diff
@@ -351,4 +351,36 @@ describe('HTTP API Integration Test', function () {
       expect(response.status).to.equal(404);
     });
   });
+
+  describe('Payload version per function', () => {
```

Comment:
I don't think we need an integration test for this one. Instead, let's configure `runServerless`-based tests here: https://github.com/serverless/serverless/blob/master/test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js

Tests should check that the `httpApi` setting on function level is overriding setting from provider.